novawallet/Common/Services/RemoteSubscription/Substrate/ProxyAccountSubscription.swift

```diff
@@ -0,0 +1,200 @@
+
+            let storageKeyFactory = StorageKeyFactory()
+
+            let codingOperation = MapKeyEncodingOperation(
+                path: path,
+                storageKeyFactory: storageKeyFactory,
+                keyParams: [accountId]
+            )
+
+            codingOperation.addDependency(codingFactoryOperation)
+
+            codingOperation.configurationBlock = {
+                do {
+                    guard let result = try codingFactoryOperation.extractResultData() else {
+                        codingOperation.cancel()
+                        return
+                    }
+
+                    codingOperation.codingFactory = result
+
+                } catch {
+                    codingOperation.result = .failure(error)
+                }
+            }
+
+            let mapOperation = ClosureOperation<Data?> { [weak self] in
+                do {
+                    return try codingOperation.extractNoCancellableResultData().first
+                } catch StorageKeyEncodingOperationError.invalidStoragePath {
+                    self?.logger?.warning("Subscription path missing in runtime: \(codingOperation.path)")
+                    return nil
+                }
+            }
+
+            mapOperation.addDependency(codingOperation)
+
+            mapOperation.completionBlock = { [weak self] in
+                do {
+                    if let remoteKey = try mapOperation.extractNoCancellableResultData() {
+                        let key = SubscriptionStorageKeys(remote: remoteKey, local: localKey)
+                        self?.subscribeToRemote(with: key)
+                    }
+                } catch {
+                    self?.logger?.error("Did receive error: \(error)")
+                }
+            }
+
+            let operations = [codingFactoryOperation, codingOperation, mapOperation]
+
+            operationQueue.addOperations(operations, waitUntilFinished: false)
+
+        } catch {
+            logger?.error("Did receive unexpected error \(error)")
+        }
+    }
+
+    private func subscribeToRemote(
+        with keyPair: SubscriptionStorageKeys
+    ) {
+        mutex.lock()
+
+        defer {
+            mutex.unlock()
+        }
+
+        do {
+            guard let connection = chainRegistry.getConnection(for: chainId) else {
+                throw ChainRegistryError.connectionUnavailable
+            }
+
+            let storageParam = keyPair.remote.toHex(includePrefix: true)
+
+            let updateClosure: (StorageSubscriptionUpdate) -> Void = { [weak self] update in
+                self?.handleUpdate(update.params.result)
+            }
+
+            let failureClosure: (Error, Bool) -> Void = { [weak self] error, unsubscribed in
+                self?.logger?.error("Did receive subscription error: \(error) \(unsubscribed)")
+            }
+
+            let subscriptionId = try connection.subscribe(
```

Comment:
It is a complex legacy way of doing subscription. It should work for now but i came up with simpler one using ```CallbackBatchStorageSubscription```. You can check it in the ```RelaychainMultistakingUpdateService```.